### DISCUSSION TITLE
Error in compiling Mfront  behavior file after installing through install.sh

### DISCUSSION THREAD

**Discussion by asingh1226** [thelfer/tfel#214] [General]
Dear all,
I was able to install  (one and half years ago) before by running "install.sh" file mentioned on the site below and there used to be no  compilation error for any MFront behavior files:
https://comet-fenics.readthedocs.io/en/latest/demo/plasticity_mfront/plasticity_mfront.py.html

However, now when I try to compile the mfront library following the command below:

mfront --obuild --interface=generic IsotropicLinearHardeningPlasticity.mfront

then I am getting following compilation error  probably because of this new version:

Treating target : all
make: *** No rule to make target '$PREFIX/codes/tfel/master/install-a4ac14ce4784ae25ba305e5edfb1014a4e86193c/include/TFEL/Material/OutOfBoundsPolicy.hxx', needed by 'IsotropicLinearHardeningPlasticity-generic.o'.  Stop.
callMake: can't build target 'all'
libraries building went wrong
Command was: make -C src -f Makefile.mfront all -s 

Any solution how to fix the compilation error.

**Comment by thelfer** [comment]
Hi @asingh1226

When upgrading to a new version, you shall remove the `include` and `src` directory that may contain links to the previous version.

However, something seems strange here as the message contains `$PREFIX`. If removing `include` and `src` does not work, would you paste the outpout of the following commands:

```bash
$ echo $TFELHOME
$ tfel-config --includes
```

**Reply by asingh1226** [reply]
Sorry, $PREFIX was meant for the $HOME directory which I wrote in the discussion. Please ignore that. 

And this was a fresh install. I had deleted everything including src and include directories.

The command $echo $TFELHOME gives
/home/amit/codes/tfel/master/install-ac173b583eade69d1d4b282f363e7def0cce36d6

**Reply by thelfer** [reply]
So, after removing the `include` and `src` directories, it still does not compile ?

**Reply by asingh1226** [reply]
No it does not compile. The installation ran smoothly. However I am getting error when compiling Mfront behavior files. The error is

Treating target : all
make: *** No rule to make target '/home/amit/softwares/mfront/codes/tfel/master/install-a4ac14ce4784ae25ba305e5edfb1014a4e86193c/include/TFEL/Material/OutOfBoundsPolicy.hxx', needed by 'IsotropicLinearHardeningPlasticity-generic.o'.  Stop.
callMake: can't build target 'all'
libraries building went wrong
Command was: make -C src -f Makefile.mfront all -s

**Reply by thelfer** [reply]
Paths are not consistent:

`echo $TFELHOME` returns: 

```bash
/home/amit/codes/tfel/master/install-ac173b583eade69d1d4b282f363e7def0cce36d67
```

while the error gives this installation path:

```bash
/home/amit/softwares/mfront/codes/tfel/master/install-a4ac14ce4784ae25ba305e5edfb1014a4e86193c
```

I am quite surprised, because mfront always takes the values of the `TFELHOME` environment variable. If you removed the `include` and `src` directories, there is no way for the other path to appear....

**Reply by asingh1226** [reply]
Sorry I have been trying too many things thats why earlier $TFELHOME was giving a different output. Earlier I had installed in a different location:  /home/amit/. Now I am installing in a different one: /home/amit/softwares/mfront

But the results are the same. Installation works fine but when I try compiling Mfront behavior files, I get errors. Would it be possible if you can install fresh in a different location than the currently installed directory by using install.sh bash script file? Perhaps you will get the same error.

**Reply by thelfer** [reply]
I do have multiple installations and it works fine as long as my environment variables are consistent, which is not the case here. We need to figure out what's wrong. Please do that:

```bash
$ rm -fr include src
$ which mfront
$ echo $TFELHOME 
$ mfront --obuild --interface=generic IsotropicLinearHardeningPlasticity.mfront
```

and send me all the outputs.

**Reply by asingh1226** [reply]
Thanks for your patience. As you asked, following are the outputs:

$ which mfront
/home/amit/softwares/mfront/codes/tfel/master/install-ac173b583eade69d1d4b282f363e7def0cce36d6/bin/mfront

$ echo $TFELHOME 
/home/amit/softwares/mfront/codes/tfel/master/install-ac173b583eade69d1d4b282f363e7def0cce36d6

$ mfront --obuild --interface=generic IsotropicLinearHardeningPlasticity.mfront
Treating target : all
make: *** No rule to make target '/home/amit/softwares/mfront/codes/tfel/master/install-a4ac14ce4784ae25ba305e5edfb1014a4e86193c/include/TFEL/Material/OutOfBoundsPolicy.hxx', needed by 'IsotropicLinearHardeningPlasticity-generic.o'.  Stop.
callMake: can't build target 'all'
libraries building went wrong
Command was: make -C src -f Makefile.mfront all -s

**Reply by asingh1226** [reply]
Somehow its working now. Most likely my env.sh file was corrupt. I deleted everything and did a fresh installation. Thank you for your response. The issue is resolved now.
